server/src/document/document.controller.ts

```diff
@@ -70,9 +70,10 @@ export class DocumentController {
     }
   }
 
+  
   @Get('/*')
-  async read(@Param() path, @Res() res) {
-    const stream = await this.documentService.getPublicPackageOrArtifactDocument(path);
+  async read(@Param() sharepointRelativePath, @Res() res) {
+    const stream = await this.documentService.getDocument(sharepointRelativePath[0]);
```

Comment:
Okay. I like what you did in documentService, where you used array destructuring to add clarity to what was being destructured. But you went backwards here, using index notation.

Reply:
@allthesignals  I tried destructing here, like `@Param() [sharepointRElativePath]`, but there's some funky stuff going on with TypeScript decorators not liking destructuring. I think the `@Param()` decorator expects something atomic passed in. I looked into ways to work around it but they seemed to add more complexity. Do you know a way?

Reply:
You don't need to destructure in the method signature. I'm not sure of a way. Have you found anything online?

Reply:
Hmm okay. I could do something like 
```
const [ firstElem] = sharepointRelativePath;
const stream = await this.documentService.getDocument(firstElem);
```
though I don't think it's necessarily clearer. I'm not a fan of code golf, but in this case I think `[0]` is more to-the-point. Just a personal preference. I don't think it's a big deal either way, though.

Reply:
Most answers online point to more complex code, and just suggest destructuring in the body to keep things simple.

Reply:
@godfreyyeung uh, code golf? What is that? Yeah, if you say "firstElem," it doesn't tell you anything about what it actually is. I'm mostly just looking for a semantic naming for the thing. Your call.

Reply:
Code golf is when programmers try to shorten code as much as possible, often to detriment of readability. Gotcha. Tbh I don't know why in this case sharepointRelativePath returns an array. The Nest docs say the `@Param` decorator should return an object?